Objects/abstract.c

```diff
@@ -819,6 +819,20 @@ binary_op(PyObject *v, PyObject *w, const int op_slot, const char *op_name)
     PyObject *result = binary_op1(v, w, op_slot);
     if (result == Py_NotImplemented) {
         Py_DECREF(result);
+
+        if (op_slot == NB_SLOT(nb_rshift) && \
+                strcmp(v->ob_type->tp_name, "builtin_function_or_method") == 0) {
```

Comment:
Fair point. Since `builtin_print` is a static symbol in `bltinmodule`, your second suggestion looks good to me.